src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -239,22 +269,928 @@ private void ImportLoadFloat(double value)
+            bool result = default(bool);
+            StackItem op1 = PopWithValidation();
+            StackItem op2 = PopWithValidation();
+
+            StackValueKind kind = (op1.Kind > op2.Kind) ? op1.Kind : op2.Kind;
+            switch (kind)
+            {
+                case StackValueKind.Int32:
+                    {
+                        int val1 = op1.AsInt32Unchecked();
+                        int val2 = op2.AsInt32Unchecked();
+
+                        switch (opcode)
+                        {
+                            case ILOpcode.ceq:
+                                result = val1 == val2;
+                                break;
+                            case ILOpcode.cgt:
+                                result = val2 > val1;
+                                break;
+                            case ILOpcode.cgt_un:
+                                result = (uint)val2 > (uint)val1;
+                                break;
+                            case ILOpcode.clt:
+                                result = val2 < val1;
+                                break;
+                            case ILOpcode.clt_un:
+                                result = (uint)val2 < (uint)val1;
+                                break;
+                            default:
+                                Debug.Assert(false);
+                                break;
+                        }
+                    }
+                    break;
+                case StackValueKind.Int64:
+                    {
+                        long val1 = op1.AsInt64Unchecked();
+                        long val2 = op2.AsInt64Unchecked();
+
+                        switch (opcode)
+                        {
+                            case ILOpcode.ceq:
+                                result = val1 == val2;
+                                break;
+                            case ILOpcode.cgt:
+                                result = val2 > val1;
+                                break;
+                            case ILOpcode.cgt_un:
+                                result = (ulong)val2 > (ulong)val1;
+                                break;
+                            case ILOpcode.clt:
+                                result = val2 < val1;
+                                break;
+                            case ILOpcode.clt_un:
+                                result = (ulong)val2 < (ulong)val1;
+                                break;
+                            default:
+                                Debug.Assert(false);
+                                break;
+                        }
+                    }
+                    break;
+                case StackValueKind.NativeInt:
+                    {
+                        IntPtr val1 = op1.AsNativeIntUnchecked();
+                        IntPtr val2 = op2.AsNativeIntUnchecked();
+#if BIT64
+                        if (!opcode.ToString().Contains("un"))
```

Comment:
`Enum.ToString()` is a very slow operation that involves reflection. This should check explicitly for the set of unsigned `opcodes` - doing so would be like 100+x faster.

Reply:
I didn't think of this. Thanks!